src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -206,3 +220,86 @@ private[hyperspace] object IndexSummary {
     indexDirPath.toString
   }
 }
+
+/**
+ * Case class representing index metadata and index statistics from latest index version.
+ *
+ * @param name index name.
+ * @param indexedColumns indexed columns.
+ * @param includedColumns included columns.
+ * @param numBuckets number of buckets.
+ * @param schema index schema json.
+ * @param indexLocation index location.
+ * @param state index state.
+ * @param kind index kind.
```

Comment:
how about `type` instead of `kind`?

Reply:
It is called `kind` in IndexLogEntry and the same term is used here.